embassy-stm32/src/usb/otg.rs

```diff
@@ -803,634 +255,52 @@ impl<'d, T: Instance> Bus<'d, T> {
-        }
+        self.inner.endpoint_is_stalled(ep_addr)
     }
 
     fn endpoint_set_enabled(&mut self, ep_addr: EndpointAddress, enabled: bool) {
-        trace!("endpoint_set_enabled ep={:?} en={}", ep_addr, enabled);
-
-        assert!(
-            ep_addr.index() < T::ENDPOINT_COUNT,
-            "endpoint_set_enabled index {} out of range",
-            ep_addr.index()
-        );
-
-        let r = T::regs();
-        match ep_addr.direction() {
-            Direction::Out => {
-                critical_section::with(|_| {
-                    // cancel transfer if active
-                    if !enabled && r.doepctl(ep_addr.index()).read().epena() {
-                        r.doepctl(ep_addr.index()).modify(|w| {
-                            w.set_snak(true);
-                            w.set_epdis(true);
-                        })
-                    }
-
-                    r.doepctl(ep_addr.index()).modify(|w| {
-                        w.set_usbaep(enabled);
-                    });
-
-                    // Flush tx fifo
-                    r.grstctl().write(|w| {
-                        w.set_txfflsh(true);
-                        w.set_txfnum(ep_addr.index() as _);
-                    });
-                    loop {
-                        let x = r.grstctl().read();
-                        if !x.txfflsh() {
-                            break;
-                        }
-                    }
-                });
-
-                // Wake `Endpoint::wait_enabled()`
-                T::state().ep_out_wakers[ep_addr.index()].wake();
-            }
-            Direction::In => {
-                critical_section::with(|_| {
-                    // cancel transfer if active
-                    if !enabled && r.diepctl(ep_addr.index()).read().epena() {
-                        r.diepctl(ep_addr.index()).modify(|w| {
-                            w.set_snak(true); // set NAK
-                            w.set_epdis(true);
-                        })
-                    }
-
-                    r.diepctl(ep_addr.index()).modify(|w| {
-                        w.set_usbaep(enabled);
-                        w.set_cnak(enabled); // clear NAK that might've been set by SNAK above.
-                    })
-                });
-
-                // Wake `Endpoint::wait_enabled()`
-                T::state().ep_in_wakers[ep_addr.index()].wake();
-            }
-        }
+        self.inner.endpoint_set_enabled(ep_addr, enabled)
     }
 
     async fn enable(&mut self) {
-        trace!("enable");
-        // TODO: enable the peripheral once enable/disable semantics are cleared up in embassy-usb
+        self.inner.enable().await
     }
 
     async fn disable(&mut self) {
-        trace!("disable");
-
-        // TODO: disable the peripheral once enable/disable semantics are cleared up in embassy-usb
-        //Bus::disable(self);
+        Bus::disable(self);
```

Comment:
there's nothing calling `Bus::enable()` or `Bus::disable()` yet in embassy-usb. Could you put back the `Bus::disable()` in drop?

Reply:
Oops it was never meant to be removed from Drop.